core/src/main/java/jenkins/slaves/EncryptedSlaveAgentJnlpFile.java

```diff
@@ -35,27 +36,23 @@
  */
 public class EncryptedSlaveAgentJnlpFile implements HttpResponse {
     /**
-     * The object that owns the Jelly view that renders JNLP file.
-     * For example {@link SlaveComputer}.
+     * The object that owns the Jelly view that renders JNLP file this is always 
+     * some type of {@link SlaveComputer}.
      */
-    private final AccessControlled it;
+    private final SlaveComputer it;
     /**
      * Name of the view that renders JNLP file that belongs to {@link #it}.
      */
     private final String viewName;
-    /**
-     * Name of the agent, which is used to determine secret HMAC code.
-     */
-    private final String slaveName;
+
     /**
      * Permission that allows plain text access. Checked against {@link #it}.
      */
     private final Permission connectPermission;
 
-    public EncryptedSlaveAgentJnlpFile(AccessControlled it, String viewName, String slaveName, Permission connectPermission) {
+    public EncryptedSlaveAgentJnlpFile(SlaveComputer it, String viewName, Permission connectPermission) {
```

Comment:
It is possible to keep backwards compatibility here if you keep the existing constructor and compute the MAC for the slave there.

In your new constructor get the mac and store it, and then use this stored mac in `generateResponse()`